Experimental/src/org/sleuthkit/autopsy/experimental/textclassifier/TextClassifierFileIngestModule.java

```diff
@@ -0,0 +1,121 @@
+import org.sleuthkit.autopsy.ingest.IngestModule;
+import org.sleuthkit.autopsy.ingest.IngestModuleReferenceCounter;
+import org.sleuthkit.autopsy.textextractors.TextExtractor.InitReaderException;
+import org.sleuthkit.autopsy.textextractors.TextExtractorFactory;
+import org.sleuthkit.autopsy.textextractors.TextExtractorFactory.NoTextExtractorFound;
+import org.sleuthkit.datamodel.AbstractFile;
+import org.sleuthkit.datamodel.BlackboardArtifact;
+import org.sleuthkit.datamodel.BlackboardAttribute;
+import org.sleuthkit.datamodel.TskCoreException;
+
+/**
+ * Classifies a file as notable or not notable, based on a model trained on
+ * labeled data.
+ */
+public class TextClassifierFileIngestModule extends FileIngestModuleAdapter {
+    private final static Logger logger = Logger.getLogger(TextClassifierFileIngestModule.class.getName());
+    private Blackboard blackboard;
+    private DocumentCategorizerME categorizer;
+    private TextClassifierUtils utils;
+
+    @Messages({"TextClassifierFileIngestModule.noClassifiersFound.subject=No classifiers found.",
+        "# {0} - classifierDir", "TextClassifierFileIngestModule.noClassifiersFound.message=No classifiers were found in {0}, text classifier will not be executed."})
+    @Override
+    public void startUp(IngestJobContext context) throws IngestModule.IngestModuleException {
+        utils = new TextClassifierUtils();
+
+        try {
+            categorizer = TextClassifierUtils.loadModel();
+        } catch (IOException ex) {
+            throw new IngestModule.IngestModuleException("Unable to load model for text classifier module.", ex);
+        }
+
+        try {
+            blackboard = Case.getCurrentCaseThrows().getServices().getBlackboard();
+        } catch (NoCurrentCaseException ex) {
+            throw new IngestModule.IngestModuleException("Exception while getting open case.", ex);
+        }
+    }
+
+    @Override
+    public ProcessResult process(AbstractFile file) {
+        if (!utils.isSupported(file, this)) {
+            return ProcessResult.OK;
+        }
+
+        if (file.getSize() > TextClassifierUtils.MAX_FILE_SIZE) {
+            //prevent it from allocating gigabytes of memory for extremely large files
+            logger.log(Level.INFO, "Encountered file " + file.getParentPath() + file.getName() + " with object id of "
+                    + file.getId() + " which exceeds max file size of " + TextClassifierUtils.MAX_FILE_SIZE + " bytes, with a size of " + file.getSize());
+            return IngestModule.ProcessResult.OK;
+        }
+
+        boolean isNotable;
+        try {
+            isNotable = classify(file);
+        } catch (TextExtractorFactory.NoTextExtractorFound ex) {
+            logger.log(Level.SEVERE, "NoTextExtractorFound in categorizing : " + ex.getMessage());
+            return ProcessResult.ERROR;
+        } catch (InitReaderException ex) {
+            logger.log(Level.SEVERE, "InitReaderException in categorizing : " + ex.getMessage());
```

Comment:
These catches should probably be doing logger.log(Level, String, Exception) instead of logger.log(Level, String) to preserve stack traces